src/metabase/util/honeysql_extensions.clj

```diff
@@ -64,6 +66,12 @@
   java.lang.Number
   (to-sql [x] (str x)))
 
+;; Ratios are represented as the division of two numbers which may cause order-of-operation issues when dealing with
+;; queries. The easiest way around this is to convert them to their decimal representations (with some rounding).
+(extend-protocol honeysql.format/ToSql
+  clojure.lang.Ratio
+  (to-sql [x] (str (u/round-to-decimals 5 x))))
```

Comment:
instead of rounding this you should just use `double` to convert it to a Double and then it will do the right thing.

e.g.

```clj
(to-sql [x] (double x))
```

Reply:
I wasn't sure. I saw rounding elsewhere so I figured I would here. I can remove it.

As to casting to strings there was a comment above it that said Numbers needed to be treated as strings so I think as long as that comment still applies a string cast is necessary.
```
;; HoneySQL 0.7.0+ parameterizes numbers to fix issues with NaN and infinity -- see
;; https://github.com/jkk/honeysql/pull/122. However, this broke some of Metabase's behavior, specifically queries
;; with calculated columns with numeric literals -- some SQL databases can't recognize that a calculated field in a
;; SELECT clause and a GROUP BY clause is the same thing if the calculation involves parameters. Go ahead an use the
;; old behavior so we can keep our HoneySQL dependency up to date.
(extend-protocol honeysql.format/ToSql
  java.lang.Number
  (to-sql [x] (str x)))
```

Reply:
for example consider a Ratio like `(/ 1 1000000)`. 

```clj
(u/round-to-decimals 5 (/ 1 1000000)) ;; -> 0.0
(double (/ 1 1000000)) ;; -> 1.0E-6
```

If you round to 5 decimal places, your number is now zero! And that's usually  a good thing -- dividing by zero is frowned upon by most DBMSes.

Reply:
Right, we pass numbers in to SQL as literals instead of prepared statement args (e.g. `SELECT 1.5` instead of `SELECT ?`) because it breaks behavior with some DBs. But that's handled by the `to-sql` implementation for Number; you need not be concerned with that. Pass the Double to `to-sql` like in my example above and and then that implementation can convert it to a literal or not as needed

Reply:
Ah I see. I wasn't sure if that other method would get called if I returned a double.

Reply:
Sorry, I think I typed my example above wrong. Do

```clj
(to-sql [x] (to-sql (double x)))
```

Reply:
All set